src/ContigInterval.js

```diff
@@ -0,0 +1,41 @@
+/* @flow */
+
+var Interval = require('./Interval');
+
+/**
+ * Class representing a closed interval on the genome: contig:start-stop.
+ *
+ * The contig may be either a string ("chr22") or a number (in case the contigs
+ * are indexed, for example).
+ */
+class ContigInterval {
+  contig: string|number;
+  interval: Interval;
+
+  constructor(contig: string|number, start: number, stop: number) {
+    this.contig = contig;
+    this.interval = new Interval(start, stop);
+  }
+
+  // TODO: make these getter methods & switch to Babel.
```

Comment:
Ah yeah, cool, makes sense. That's definitely possible in ES5 & it prevents the JIT from kicking, in just FYI.